plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -264,7 +264,12 @@ public Optional<ProjectionApplicationResult<ConnectorTableHandle>> applyProjecti
                 return Optional.empty();
             }
 
-            verify(tableColumnSet.containsAll(newColumnSet), "applyProjection called with columns %s and some are not available in existing query: %s", newColumnSet, tableColumnSet);
+            Set<JdbcColumnHandle> tableSyntheticColumnSet = ImmutableSet.<JdbcColumnHandle>builder()
```

Comment:
Is it possible to provide a test?

Reply:
@raunaqmorarka @sopel39 could you please advice which connector invokes `PruneTableScanColumns#pruneColumns` e.g. running `BaseConnectorTest#testAddAndDropColumnName`

Reply:
That rule should trigger whenever there are more columns output from the table scan than are used by the rest of the query plan. E.g. `select nationkey, count(*) from (select * from nation) group by 1` in tpch.tiny

Reply:
@ssheikin it looks like a bug fix. do you have a query reproducing the problem?